env_functions.sh

```diff
@@ -29,9 +29,12 @@ hg_prompt() {
   local repo
   repo=$(prompt::_find_repo .hg) || return 0
   cd "$repo" || return # so Mercurial doesn't have to do the same find we just did
-  local branch num_heads heads
```

Comment:
Please keep these `local` declarations as-is, the `local num_heads=$(...) || return 0` line will not work as intended; see [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155).

Reply:
Thank you for hand-holding here! Learned so much from this PR attempt.. :D

Reply:
My pleasure, thanks for iterating with me.